### PR TITLE
added support for ROS 2 Iron

### PR DESCRIPTION
**Pull request by razr**

### CODE REVIEW

**Review by LanderU — APPROVED** [review]
LGTM!

**Comment by carlossvg** [comment]
Script to generate release plot: https://github.com/ros-realtime/ros_realtime_benchmarks_config/tree/main/cyclictest

File: image_builder/data/jammy-rt-iron/rootfs/etc/apt/sources.list.d/ros2.list
Comment (nightduck) [inline image_builder/data/jammy-rt-iron/rootfs/etc/apt/sources.list.d/ros2.list:1]: This link should be changed to

    deb [arch=arm64 signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu/ iron main

The link will be valid Tuesday, May 23rd, after Iron releases

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,9 @@ focal-rt-ros2:
 jammy-rt-ros2:
 	sudo ./ros-rt-img build jammy-rt jammy-rt-humble
 
+jammy-rt-iron:
```
Comment (shuhaowu) [inline Makefile:11]: Can you add this target to `.PHONY`?

**Review by shuhaowu — APPROVED** [review]
Do you want to add this to `.github/workflows/build.yml` so we can build it on Github CI as well?

**Comment by shuhaowu** [comment]
Latency analysis scripts: https://github.com/ros-realtime/ros-realtime-rpi4-image/tree/master/latency-analysis

File: image_builder/data/jammy-rt-iron/rootfs/etc/apt/sources.list.d/ros2.list [hunk 2]
```diff
@@ -0,0 +1 @@
+deb [arch=arm64 signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2-testing/ubuntu jammy main
```
Comment (LanderU) [inline image_builder/data/jammy-rt-iron/rootfs/etc/apt/sources.list.d/ros2.list:1]:
```suggestion
deb [arch=arm64 signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu jammy main
```

**Review by LanderU — CHANGES_REQUESTED** [review]

**Comment by shuhaowu** [comment]
@razr I'm closing  this as we discussed to move towards Jazzy + Noble.

**Pull request closed without merging**